### PR TITLE
Added apache spark docs

### PR DESCRIPTION
**Pull request by mzitnik**

### CODE REVIEW

File: docs/en/integrations/index.mdx [hunk 1]
```diff
@@ -230,7 +230,7 @@ We are actively compiling this list of ClickHouse integrations below, so it's no
...
 |Apache SkyWalking|<img src={require('./images/logos/skywalking_logo.jpeg').default} class="image" alt="SkyWalking logo" style={{width: '3rem', 'background-color':'transparent'}}/>|Data management|Open-source APM system that provides monitoring, tracing and diagnosing capabilities for distributed systems in Cloud Native architectures.|[Blog](https://skywalking.apache.org/blog/2024-03-12-monitoring-clickhouse-through-skywalking/)|
-|Apache Spark|<SparkSVG style={{width: '3rem'}} />|Data ingestion|Spark ClickHouse Connector is a high performance connector built on top of Spark DataSource V2.|[GitHub](https://github.com/housepower/spark-clickhouse-connector),<br/>[Documentation](https://housepower.github.io/spark-clickhouse-connector/)|
+|Apache Spark|<SparkSVG style={{width: '3rem'}} />|Data ingestion|Spark ClickHouse Connector is a high performance connector built on top of Spark DataSource V2.|[GitHub](https://github.com/housepower/spark-clickhouse-connector),<br/>[Documentation](/docs/en/integrations/data-ingestion/apache-spark/index.md)|
```
Comment (mshustov) [inline docs/en/integrations/index.mdx:233]: Should be moved to Core integrations

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+# Get the Library
+
+## Download the Library
+
+The name pattern of binary jar is
+```
+clickhouse-spark-runtime-${spark_binary_version}_${scala_binary_version}-${version}.jar
+```
+you can find all available released jars under [Maven Central Repository](https://repo1.maven.org/maven2/com/github/housepower)
```
Comment (mshustov) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:20]: are we going to update the Maven account later?

**Review by mshustov — COMMENTED** [review]
we need to move [Requirements](https://github.com/ClickHouse/spark-clickhouse-connector?tab=readme-ov-file#requirements) and [Compatibility Matrix](https://github.com/ClickHouse/spark-clickhouse-connector?tab=readme-ov-file#compatible-matrix) from Readme

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+description: Introduction to Apache Spark with ClickHouse
+---
+
+# Integrating Apache Spark with ClickHouse
+
+[Apache Spark](https://spark.apache.org/) Apache Spark™ is a multi-language engine for executing data engineering, data science, and machine learning on single-node machines or clusters.
+
+# Get the Library
+
+## Download the Library
```
Comment (mshustov) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:14]: Do we need to add a list of supported types? Complex types might be tricky https://github.com/ClickHouse/clickhouse-java/issues/1042

File: docs/en/integrations/index.mdx (hunk 1 above)
Comment (mzitnik) [reply docs/en/integrations/index.mdx:233]: Yep you are right missed it

File: docs/en/integrations/data-ingestion/apache-spark/index.md (hunk 2 above)
Comment (mzitnik) [reply docs/en/integrations/data-ingestion/apache-spark/index.md:20]: Yes

**Review by mshustov — APPROVED** [review]

**Pull request merged**